### PR TITLE
Add macos_user_defaults resource from mac_os_x cookbook

### PR DESCRIPTION
**Pull request by tas50**
Brought in as-is from the upstream cookbook

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

**Comment by thommay** [comment]
Guess I should say explicitly this needs all the documentation goodness

**Comment by tas50** [comment]
I figured we'd wait for the results of our docs team meeting before added all of that

File: lib/chef/resource/macos_userdefaults.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+               description: "",
+               default: false,
+               desired_state: false
+
+      action :write do
+        description "Write the setting to the specified domain"
+
+        @userdefaults = Chef::Resource.resource_for_node(:macos_userdefaults, node).new(new_resource.name)
+        @userdefaults.key(new_resource.key)
+        @userdefaults.domain(new_resource.domain)
```
Comment (lamont-granquist) [inline lib/chef/resource/macos_userdefaults.rb]:
```ruby
userdefault = build_resource(:macos_userdefaults, new_resource.name) do
  key new_resource.key
  domain new_resource.domain
end
```

**Comment by tas50** [comment]
Double checked this with lamont via slack. Merging on the verbal approval

**Pull request merged**